Common/src/main/java/me/egg82/tfaplus/services/InternalAPI.java

```diff
@@ -527,79 +546,76 @@ public boolean seekHOTPCounter(UUID uuid, String[] tokens, JedisPool redisPool,
 
         // SQL
         HOTPData result = null;
-        try {
-            if (sqlType == SQLType.MySQL) {
-                result = MySQL.updateHOTP(sql, storageConfigNode, uuid, data.getLength(), counter, data.getKey()).get();
-            } else if (sqlType == SQLType.SQLite) {
-                result = SQLite.updateHOTP(sql, storageConfigNode, uuid, data.getLength(), counter, data.getKey()).get();
-            }
-        } catch (ExecutionException ex) {
-            logger.error(ex.getMessage(), ex);
-        } catch (InterruptedException ex) {
-            logger.error(ex.getMessage(), ex);
-            Thread.currentThread().interrupt();
-        }
+        result = getHotpData(uuid, data.getLength(), counter, data.getKey(), result);
 
         if (result == null) {
             return false;
         }
 
         // Redis
-        Redis.update(result, redisPool, redisConfigNode);
+        Redis.update(result, ConfigUtil.getRedisPool(), ConfigUtil.getRedisConfigNode());
 
         // Rabbit
-        RabbitMQ.broadcast(result, rabbitConnection);
+        broadcastResult(result);
 
         // Cache
         hotpCache.put(uuid, new HOTPCacheData(data.getLength(), counter, data.getKey()));
 
         return true;
     }
 
-    public boolean hasAuthy(UUID uuid, JedisPool redisPool, ConfigurationNode redisConfigNode, Connection rabbitConnection, SQL sql, ConfigurationNode storageConfigNode, SQLType sqlType, boolean debug) {
-        if (debug) {
+    private void broadcastResult(HOTPData result) {
+        try {
+            RabbitMQ.broadcast(result, getRabbitConnection());
```

Comment:
Not using the broadcast method, here?
Redis & Rabbit broadcast/update should probably be in one method, if we're going to make it a method anyway.